docs/cli.rst

```diff
@@ -37,30 +37,74 @@ The builtin REPL supports basic code completion suggestions, syntax highlighting
+
+1. Ensure that a ``basilisp.edn`` file is present at the root of your project, even if it is empty.
+2. Run the ``Jack-in`` command in your editor and select ``Basilisp`` if prompted. The Editor should then start the server and connect to it.
+
+Start from an editor with a Clojure extension not yet supporting Basilisp
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. note::
+   The extension's ``connect`` command is used to connect your editor to a running nREPL server. It requires the port number where the server is running at.
+
+.. note::
+   Basilisp code files use the ``.lpy`` filename suffix. You might need to adjust your editor's extension settings to recognize this suffix as a Clojure file. Alternatively, you can evaluate code inside ``.clj`` files, though importing these files from other Basilisp files might not be possible due to the different file extension.
+
+If your editor extension does not yet support Basilisp, or if you prefer more control over the nREPL server, you can start the server from the command line and connect to using your extension's ``connect`` command.
+
+The nREPL server when started will provide the host and port number it is listening on.
+
+To view available command line options, use
+
+.. code-block:: bash
+
+   basilisp nrepl-server -h
+
+
+To start the server on a random port, use
 
 .. code-block:: bash
 
    basilisp nrepl-server
    # => nREPL server started on port 50407 on host 127.0.0.1 - nrepl://127.0.0.1:50407
 
-You can then establish a connection from your IDE to the server address.
 
-- from `Emacs`, using `CIDER <https://github.com/clojure-emacs/cider>`_
+To start the server on a specific port, use
+
+.. code-block:: bash
 
-  .. code-block::
+   basilisp nrepl-server --port 8889
+   #=> nREPL server started on port 8889 on host 127.0.0.1 - nrepl://127.0.0.1:8889
 
-     M-x cider-connect-clj
+Some extensions can connect to a running server automatically by looking for a ``.nrepl-port`` file at the root of the project. This file contains the port the server is listening on.
 
-- from `Visual Studio Code`, using `Calva <https://calva.io/>`_
+The ``nrepl-server`` command will generate this file in the current working directory where the server is started. If your extension's ``connect`` command looks for this file, run the server command from the root of the project, so that is generated in there
 
-  .. code-block::
+.. code-block:: bash
+
+   cd <project-root-directory>
+   basilisp nrepl-server
+   #=> nREPL server started on port 632128 on host 127.0.0.1 - nrepl://127.0.0.1:63128
+
+Alternatively, specify the full path where this file should be generated
+
+.. code-block:: bash
 
-     REPL -> Connect to a running REPL in your project -> Generic
+   basilisp nrepl-server --port-filepath <project-root-directory>/.nrepl-port
+   #=> nREPL server started on port 62079 on host 127.0.0.1 - nrepl://127.0.0.1:62079
 
-.. _run_basilisp_code:
```

Comment:
Looks like this anchor might've gotten accidentally removed. Can you add it back?

Reply:
Restored, sorry about this